Backend/Backend/Controllers/PodcastController.cs

```diff
@@ -398,6 +402,56 @@ public async Task<ActionResult> GetEpisodeThumbnail(Guid podcastId, Guid episode
         }
     }
 
+    /// <summary>
+    /// THis function saves the last watched position on a specific episode
+    /// On the frontend:
+    ///     - You need to add a onBeforeUnload  hook to the episode webpage and this hook should
+    ///       send request to this route.
+    /// </summary>
+    /// <param name="podcastId"></param>
+    /// <param name="episodeId"></param>
+    /// <param name="request"></param>
+    /// <returns></returns>
+    [HttpPost("{episodeId}/saveWatchHistory")]
+    public async Task<IActionResult> SaveWatchHistory(Guid podcastId, Guid episodeId, [FromBody] EpisodeHistorySaveRequest request)
+    {
+        try
+        {
+            User? user = await _authService.IdentifyUserAsync(HttpContext);
+            if (user is null)
+                return NotFound("User not found");
+            
+            var episode = await _podcastService.GetEpisodeByIdAsync(episodeId, GetDomainUrl(HttpContext));
+            
+            // Check if user had episode interaction before
+            var interaction = await _podcastService.GetUserEpisodeInteraction(user, episode.Episode);
+            if (interaction is null)
+            {
+                interaction = new UserEpisodeInteraction(_db)
+                {
+                    EpisodeId = episode.Id,
+                    UserId = user.Id,
+                    DateListened = DateTime.Now,
+                    LastListenPosition = Math.Min(episode.Episode.Duration, request.ListenPosition)
+                };
+                await _db.UserEpisodeInteractions!.AddAsync(interaction);
+            }
+            else
+            {
+                interaction.DateListened = DateTime.Now;
+                interaction.LastListenPosition = Math.Min(episode.Episode.Duration, request.ListenPosition);
```

Comment:
The db interactions should be included inside a service instead of a controller.